### ISSUE TITLE
Spatie Media Library - Image not uploading during filament resource create page but works on edit page

### ISSUE THREAD

**Issue by ashokrajc87** [laravel-filament/filament#6143]
### Package

filament/spatie-laravel-media-library-plugin

### Package Version

v2.17.21

### Laravel Version

v9.52.5

### PHP Version

PHP 8.1.6

### Problem description

In Laravel Filament Admin Panel  -  I tried to create resources with Spatie Media Library,  the resource was created without an image and it doesn't show any errors on the logs, but it works on the edit page of the resource...

### Expected behavior

Need solution for my problem

### Steps to reproduce

Resource created with image on Laravel filament admin panel using Spatie Media Library

### Reproduction repository

https://github.com/ashokrajc87/newsportal

### Relevant log output

```shell
Spatir Media Library - Image Not Uploading On Filament Admin Panel (Resource Page)
```

**Comment by danharrin** [comment]
Please check if it is solved by #6141?

**Comment by Yo1L** [comment]
@danharrin Is storeFileName more relevant in this case ?
You can reproduce this error on your demo site.

**Comment by danharrin** [comment]
Storing the file name is not relevant here either. Please open a new issue

**Comment by Yo1L** [comment]
@danharrin no need to create a new one.
We have exactly the same problem : File not uploading during on create page but works on edit page !

**Comment by inerba** [comment]
Downgrading filament/spatie-laravel-media-library-plugin to version 2.17.20 temporarily solved the issue for me.

**Comment by Yo1L** [comment]
@inerba Thanks, it works again !

**Comment by danharrin** [comment]
@inerba can you try applying the fix from #6141 in /vendor?

**Comment by SanderCokart** [comment]
Same issue

**Comment by danharrin** [comment]
@SanderCokart did you test out the fix in #6141?

**Comment by SanderCokart** [comment]
@danharrin That fixed it!

**Comment by inerba** [comment]
> @inerba can you try applying the fix from #6141 in /vendor?

I can confirm that the fix also works in my case 🎉
